### PR TITLE
custom properties with event

### PR DESCRIPTION
**Pull request by chrisssssy**
added event for custom properties (extras) in gltf - #90

### CODE REVIEW

**Comment by atteneder** [comment]
Hi @chrisssssy and thanks a lot for this contribution.

I'll have an in-depth look soonish, but in order to approve this I need performance tests that measure the impact on JSON parsing (and potentially some unit tests covering corner cases). Depending on the result this PR can be merged as-is or it requires a scripting define to enable the additional parsing of `extras`. I'll schedule this on my side, but feel free to take it on yourself.

Other idea: the `extras` property could be added to elements besides nodes. I'll look into this as well

Regards

File: Runtime/Scripts/GltfImport.cs [hunk 1]
```diff
@@ -49,6 +49,10 @@ namespace GLTFast {
     using Loading;
 
     public class GltfImport : IGltfReadable, IGltfBuffers {
+    
+
```
Comment (atteneder) [inline Runtime/Scripts/GltfImport.cs:53]: Two concerns:
1. I'm not sure Actions/Events are the best approach to inject custom loading behavior. Up until now I encouraged people to implement custom IInstantiators (or derive from GameObjectInstantiator) to achieve this. Granted, `IGltfReadable` and `IInstantiator` need to be extended to get access to the `extras`, but that's probably cleaner.
2. This creates a dependency to GameObjects and GameObjectInstantiator. It would not work with a DOTS workflow. This is a blocker.

File: Runtime/Scripts/GltfImport.cs [hunk 2]
```diff
@@ -1571,6 +1575,12 @@ enum ChunkFormat : uint
                 }
                 
                 instantiator.SetNodeName(nodeIndex,goName);
+                
+                if (node.extras != null)
+                {
+                    GameObject go = ((GameObjectInstantiator)instantiator).GetGameobject(nodeIndex);
```
Comment (atteneder) [inline Runtime/Scripts/GltfImport.cs:1581]: This will cause a null pointer dereference exception in DOTS projects

**Review by atteneder — CHANGES_REQUESTED** [review]
See blocking problems in the line comments.
Overall I think this PR would be a make-shift solution before a better JSON parsing allows for better extras and custom data support.

**Comment by AbrarZShahriar** [comment]
Any way to get this workinng in ver 5? Even if makeshift.
